### ISSUE TITLE
Difference of results by changing settings for recomputing residuals in FindTransferAnchors

### ISSUE THREAD

**Issue by nicodemus88** [satijalab/seurat#6662]
Hi Seurat developers @andrewwbutler @yuhanH @mojaveazure @timoast and everyone, 

I am trying to perform reference mapping of my dataset to the PBMC reference dataset. However, I got very different results by adjusting the parameter `recompute.residuals` between `TRUE` or `FALSE`. I am not really clear regarding the effects or function of recomputing the Pearson residuals. 

What I understood was if it was set to `TRUE`, the Pearson residuals of my query will be calculated based on the SCT model parameters of the reference and if set to `FALSE`, the Pearson residuals originally calculated when performing SCT transformation on the dataset will be used. Am I correct here? How does this affect the label transfer and mapping? 

As you can see the results below, I got very contrasting cell labels by tweaking this parameter and when set to FALSE, I got almost all different subtypes present. But my sample are PBMCs which have been negatively selected for CD4 T cells only. I think the results from setting to TRUE is reasonable, but my boss is skeptical as the proportions are a bit off, while the one set to FALSE, the proportions of the CD4 subsets are much more reflective of the normal range in humans. Thus, I am confused on which results to accept.

In short, could someone please explain to me the difference of adjusting the recompute.residuals parameter, how it affects the results and what is the proper way to use this parameter. Thank you very much.

![Picture1](https://user-images.githubusercontent.com/24598265/200984692-009a0506-f580-4fb2-bb87-4e2691af4182.png)

**Comment by yuhanH** [comment]
hi @nicodemus88 
In your case, you should set use the default setting, `recompute.residuals = TRUE`. 
```
What I understood was if it was set to TRUE, the Pearson residuals of my query will be calculated based on the SCT model parameters of the reference and if set to FALSE, the Pearson residuals originally calculated when performing SCT transformation on the dataset will be used.
```
You're correct. This subtype query problem is primarily addressed by computing residuals based on the reference SCT model. The mapped results are strange if you use the residuals computed from your dataset because the variance of residuals can only reflect the difference between CD4 T cells and excludes the difference between all other cell types.
It is necessary for your input data to have a similar cell type proportion to the reference for "recompute.residuals = FALSE" to work.
